lib/msf/core/exploit/check_module.rb

```diff
@@ -43,7 +43,7 @@ def check
     res = mod.run_simple(
       'LocalInput'  => user_input,
       'LocalOutput' => user_output,
-      'Options'     => datastore.to_h.slice('RHOSTS', 'RHOST', 'RPORT')
+      'Options'     => datastore # XXX: This clobbers the datastore!
```

Comment:
@wvu-r7 Is there any additional context on this comment? I'm assuming this just means the check module might mutate the original datastore, which in theory could be problematic? Or is there something more subtle here? 👀

Reply:
Pretty sure it clobbers (replaces) the check module's datastore with the exploit's. The datastore options we sliced out weren't sufficient, so we took all of them.

Reply:
I haven't noted any mutation issues, but that doesn't mean there aren't any.